### DISCUSSION TITLE
When calling two OpenAI Assistant v2 operations in async, wrong assistant_id gets used

### DISCUSSION THREAD

**Discussion by faratech** [openai/openai-python#1679] [Q&A]
I am calling an intensive operation in async using FastAPI to tag a bunch of content. When I go to run another script and utilize the same FastAPI endpoint, the JSON I get back is a response to the request I sent,  but the payload/response shows the assistant_id used was the one I am calling for this other operation, even though I am explicitely sending the assistant_id to the FastAPI endpoint. When I discontinue my job running in the background the issue stops. This tells me that if I run multiple operatons in AsyncOpenAI concurrently, they will ultimately get confused running in concurrence. Does AsyncOpenAI client internally manages sessions in a way that isolates requests or will it re-use sessions across requests?

**Comment by TruthStalker** [comment]
I don't understand all of the programming lingo, I am hoping that I'm understanding your question, as I do have a lot of recent experience with us. The ChatGPT AI only does one maybe two requests at once. You have to build one on top of another. If you give it multiple instructions, it gets confused. If you ask it to analyze all data that you have built? Question leading it where you're trying to get, depending on which AI function you use, it typically will get lost and cannot do all of it and will misinterpret  your commands. The excelAI and humanized seem to be the best under ChatGPT.
